internal/cloudconfig/cloudinit/cloudinit.go

```diff
@@ -211,6 +212,22 @@ func (cfg *cloudConfig) Packages() []string {
 	return packs
 }
 
+// AddSnap is defined on the SnapConfig interface.
+func (cfg *cloudConfig) AddSnap(snap string) {
+	cfg.attrs["snaps"] = append(cfg.Snaps(), snap)
```

Comment:
Just in case I wasn't clear, we can do this via cloud-init as well via: https://cloudinit.readthedocs.io/en/latest/reference/yaml_examples/snap.html#cce-snap

Reply:
Yeah I know, but I think we serialise all the `cfg.attrs` to cloud-init. So in the cloud-init I think there is a `snaps` top level attribute which cloud-init ignores and there is the commands we rendered out in the `runcmd`/`bootcmd`.

My point is: maybe don't put it in the attrs and just have a struct variable to hold the snaps.

Reply:
> Just in case I wasn't clear, we can do this via cloud-init as well via: https://cloudinit.readthedocs.io/en/latest/reference/yaml_examples/snap.html#cce-snap

Yeah this or https://cloudinit.readthedocs.io/en/latest/reference/yaml_examples/package_update_upgrade.html#

Both will make sure snap exists and is ready AFAIK. Then we don't need to do the `snap wait core seed.loaded && break` garbage.

Reply:
But we need to eventually stop using bash to do the init and just use cloud-init everywhere.

Reply:
Ah, but I don't think we should use this package/library to implement this. We should just throw all of this away it feels incorrect.